CRM/Contact/BAO/Contact.php

```diff
@@ -306,6 +306,11 @@ public static function &create(&$params, $fixAddress = TRUE, $invokeHooks = TRUE
           }
         }
       }
+
+      // CRM-21041: set default 'Communication Style' if unset when creating a contact.
+      if (empty($params['communication_style_id'])) {
+        $params['communication_style_id'] = array_pop(CRM_Core_OptionGroup::values('communication_style', TRUE, NULL, NULL, 'AND is_default = 1'));
```

Comment:
in some versions of php this won't work because array_pop() can't be called on a function.